Explorer/Assets/DCL/EmojiPanel/EmojiPanelController.cs

```diff
@@ -0,0 +1,150 @@
+using Cysharp.Threading.Tasks;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.UI;
+using Utility;
+using Object = UnityEngine.Object;
+
+namespace DCL.Emoji
+{
+    public class EmojiPanelController
+    {
+        public event Action<string> OnEmojiSelected;
+        private readonly EmojiPanelView view;
+        private readonly EmojiPanelConfigurationSO emojiPanelConfiguration;
+        private readonly EmojiButton emojiButtonPrefab;
+        private readonly EmojiSectionView emojiSectionPrefab;
+        private readonly EmojiSearchController emojiSearchController;
+
+        private readonly List<EmojiSectionView> emojiSectionViews = new ();
+        public readonly Dictionary<string, EmojiData> EmojiNameMapping = new ();
+        private readonly Dictionary<string, string> emojiValueMapping = new ();
+        private readonly Dictionary<EmojiSectionName, RectTransform> sectionTransforms = new ();
+
+        private CancellationTokenSource cts = new CancellationTokenSource();
+
+        private int startDec;
+        private int endDec;
+        private int emojiCode;
+        private string emojiChar;
+
+        public EmojiPanelController(
+            EmojiPanelView view,
+            EmojiPanelConfigurationSO emojiPanelConfiguration,
+            TextAsset emojiMappingJson,
+            EmojiSectionView emojiSectionPrefab,
+            EmojiButton emojiButtonPrefab)
+        {
+            this.view = view;
+            this.emojiPanelConfiguration = emojiPanelConfiguration;
+            this.emojiSectionPrefab = emojiSectionPrefab;
+            this.emojiButtonPrefab = emojiButtonPrefab;
+            emojiSearchController = new EmojiSearchController(view.searchPanelView, view.emojiSearchedContent, emojiButtonPrefab);
+            emojiSearchController.OnSearchTextChanged += OnSearchTextChanged;
+            emojiSearchController.OnEmojiSelected += emoji => OnEmojiSelected?.Invoke(emoji);
+            foreach (var emojiData in JsonConvert.DeserializeObject<Dictionary<string, string>>(emojiMappingJson.text))
```

Comment:
I have a feeling that this "heavy" initialization part (and configuration) should be moved to the `plugin` and `EmojiPanelController` should be constructed from the plugin and passed to the `ChatController`.
This would sanitize this class from the view-unrelated logic

Reply:
Yeah it would be ideal but I have a dependency on the EmojiPanelView that lives inside the chat view, that I don't really have access to at plugin level considering the MVC flow